errors/errors_test.go

```diff
@@ -2,11 +2,50 @@ package errors
 
 import (
 	"fmt"
+	"strings"
 	"testing"
 
+	"github.com/iov-one/weave/weavetest/assert"
 	"github.com/pkg/errors"
 )
 
+func TestMultiErr(t *testing.T) {
+	cases := map[string]func(t *testing.T){
+		"Named error": func(t *testing.T) {
+			name := "Test"
+			err := MultiAddNamed(name, ErrEmpty)
+			assert.Equal(t, err.Named(name), ErrEmpty)
+			assert.Equal(t, err.Named("random"), nil)
+		},
+		"Named error override": func(t *testing.T) {
+			name := "Test"
+			err := MultiAddNamed(name, ErrEmpty)
+			err.AddNamed(name, ErrState)
+			assert.Equal(t, err.Named(name), ErrState)
+		},
+		"ABCICode is consistent with that of a normal error": func(t *testing.T) {
+			err := MultiAdd(ErrEmpty)
+			assert.Equal(t, err.(coder).ABCICode(), ErrEmpty.ABCICode())
+		},
+		"Error() works as expected": func(t *testing.T) {
+			err := MultiAdd()
+			assert.Equal(t, err.Error(), "")
+
+			err.Add(ErrEmpty)
+			assert.Equal(t, strings.Contains(err.Error(), ErrEmpty.Error()), true)
+
+			err.Add(ErrState)
+			assert.Equal(t, strings.Contains(err.Error(), ErrEmpty.Error()), true)
+			assert.Equal(t, strings.Contains(err.Error(), ErrState.Error()), true)
+			assert.Equal(t, strings.Contains(err.Error(), "2"), true)
+		},
+	}
+
+	for testName, tc := range cases {
+		t.Run(testName, tc)
```

Comment:
This is a really weird way of writing tests. Why not just write a separate function for each case? Those cases don't share any code (setup, common check etc). They are precisely a collection of test functions.

Reply:
I see no difference to be honest, saves some boilerplate and has a similar feel to other tests.